### ISSUE TITLE
EVSS Migration - 526 Submit - revisit EVSS To Lighthouse form 526 transformation service

### ISSUE THREAD

**Issue by RakshindaAslam** [department-of-veterans-affairs/va.gov-team#65051]
## Issue Description
 Make EVSS formatted submission into a LH formatted submission
Work in scope:

- [x] ATTRIBUTE NAMES
- [x] - currentlyVAEmployee -> currentVAEmployee: we named the attribute wrong (or it changed)
- [ ] REQUIRED BUT NULL
- [x] - /disabilities/{n}/disabilityActionType: this is required and we are sending `null`
- [x] - /disabilities/{n}/classificationCode: we are sending null for some reason (not sure if this is required)
- [x] -- this is not required. we're good.
- [ ] - /disabilities/0/serviceRelevance: this is required but we are sending `null`
- [x] - /disabilities/{n}/secondaryDisabilities/{n}/disabilityActionType: this is required but we are sending `null`. and it actually just needs to be hard-coded to `"SECONDARY"`
- [x] BAD VALUE FORMAT
- [x] - /servicePay/retiredStatus: all caps the value. check enums
- [x] DATES
- [x] - /changeOfAddress/dates/beginDate: convert to YYYY-MM-DD 
- [x] - /changeOfAddress/dates/endDate: convert to YYYY-MM-DD
- [x] - /serviceInformation/servicePeriods/{n}/activeDutyBeginDate: convert to YYYY-MM-DD
- [x] - /serviceInformation/confinements/{n}/approximateBeginDate: convert to YYYY-MM-DD
- [x] - /serviceInformation/confinements/{n}/approximateEndDate: convert to YYYY-MM-DD
- [x] - /servicePay/separationSeverancePay/datePaymentReceived: convert to YYYY-MM-DD
- [x] NULLABLE BOOLS
- [x] - /serviceInformation/reservesNationalGuardService/receivingInactiveDutyTrainingPay (another double error): change this to a `string`: "YES", "NO", or null
- [x] - /serviceInformation/servedInActiveCombatSince911 (double error): change this to a `string`: "YES", "NO", or null
- [x] - /servicePay/receivingMilitaryRetiredPay: change this to a `string`: "YES", "NO", or null
- [x] - /servicePay/receivedSeparationOrSeverancePay (double error): change this to a `string`: "YES", "NO", or null
- [x] - /servicePay/futureMilitaryRetiredPay: change to a string "YES", "NO", or null

## new LH field changes to implement

- [x] claimDate - commented on questions doc
- [x] veteranNumber - might have changed, re-check when LH API spec has been reformatted; _aurora: all good here_
- [x] mailingAddress + changeOfAddress - looks like it has addressLine1, addressLine2, AND addressLine3 now!! Need to reverse our concat logic and make field names match w/EVSS.
- [x] changeOfAddress { dates { beginningDate, endingDate } are now { beginDate, endDate }
- [x] homeless { riskOfBecomingHomeless { otherLivingSituation } is now otherDescription
- [x] federalActivation - brand new field/section, need to confirm what the fields there really are
- [x] reservesNationalGuardService { obligationTermsOfService { startDate, endDate } are now { beginDate, endDate }

---
## Tasks
- [ ] Do all this work
- [ ] Data fields are transformed correctly
- [ ] Communicate with LH on contract clarification/updates
- [ ] Depending on LH answer about optional fields, we may. need to remove optional fields that are null from the our data transfer object

## Acceptance Criteria
- [ ] Data fields in scope of this ticket are transformed correctly
- [ ] Take the transformed EVSS format and send to LH for a successful submission
    - [ ] Successful submission will depend on what is outstanding after communication with LH and what is completed in this ticket
- [ ] Identify what else needs to be transformed and create another ticket, if needed
- [ ] 

## Results/Notes
- [per a few conversations with Lighthouse ](https://dsva.slack.com/archives/C02CQP3RFFX/p1698944326755649) we will have to wait until the submit endpoint is complete and tested in order to finish this up
-- there is an outstanding question about `disability/{n}/serviceRelevance` being required
- however, we did go in and [clean up quite a few mappings and write a few more tests on our side](https://github.com/department-of-veterans-affairs/vets-api/pull/14393)
- due to the "lottery factor", [a tool was made to test lighthouse submissions](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/tree/master/teams/benefits/scripts/526/TREX/DEBUG)
- a new ticket has been created to re-revisit the transformation service: #69080

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by RakshindaAslam** [comment]
Was put on Hold as the LH schema is not yet final. But can start working on unblocked items from the ticket.

**Comment by RakshindaAslam** [comment]
Also waiting on LH to remove validation from Nullable fields
